### PR TITLE
Add pattern-category (fate#323785)

### PR DESCRIPTION
**Pull request by lnussel**

### CODE REVIEW

File: add_product_susedata [hunk 1]
```diff
@@ -337,13 +338,23 @@ for my $language (sort(@languages)) {
...
+    if ($pack->{'format'}->{'rpm:provides'}->{'rpm:entry'}) {
+      for $p (@{$pack->{'format'}->{'rpm:provides'}->{'rpm:entry'}}) {
+          next unless $p->{'name'} eq 'pattern-category()';
+          $category = uri_unescape($p->{'ver'});
+          break; # we expect only one category
+      }
+    }
     my $i18n_summary     = gettext($summary);
     my $i18n_description = gettext($description);
+    my $i18n_category    = gettext($category) if $category;
```
Comment (mlschroe) [inline add_product_susedata]: This is wrong, you must not use a trailing "if" with a my statement. It does not do what you expect.
Change this to:
```
my $i18n_category    = $category ? gettext($category) : undef; 
```

**Review by mlschroe — CHANGES_REQUESTED** [review]
See inline comment

**Comment by mlschroe** [comment]
I also vind it very strange to have a translated pattern-category in the susedata.lang file but no pattern-category in susedata. Could you please change the code so that the untranslated category is also added?

**Comment by lnussel** [comment]
done. Note there is another inconsistency, not introduced by me though. The eula tag is without lang tag in in susedata.lang

**Review by mlschroe — APPROVED** [review]

**Comment by mlschroe** [comment]
Yeah, the eula should also get a lang tag. This needs a change in ABXML::susedata_i18n, though. Anyway, it's a different issue.

**Comment by lnussel** [comment]
so anyone going to merge this and submit a package?

**Pull request merged**